pkg/service/storage/ucan.go

```diff
@@ -172,6 +179,100 @@ func NewUCANServer(storageService Service, options ...server.Option) (server.Ser
 				},
 			),
 		),
+		server.WithServiceMethod(
+			replica.AllocateAbility,
+			server.Provide(
+				replica.Allocate,
+				func(cap ucan.Capability[replica.AllocateCaveats], inv invocation.Invocation, iCtx server.InvocationContext) (replica.AllocateOk, fx.Effects, error) {
+					//
+					// UCAN Validation
+					//
+
+					// only service principal can perform an allocation
+					if cap.With() != iCtx.ID().DID().String() {
+						return replica.AllocateOk{}, nil, NewUnsupportedCapabilityError(cap)
+					}
+
+					//
+					// end UCAN Validation
+					//
+
+					// create the transfer invocation: an fx of the allocate invocation receipt.
+					trnsfInv, err := replica.Transfer.Invoke(
+						storageService.ID(),
+						storageService.ID(),
+						storageService.ID().DID().GoString(),
+						replica.TransferCaveats{
+							Space:    cap.Nb().Space,
+							Blob:     cap.Nb().Blob,
+							Location: cap.Nb().Location,
+							Cause:    inv.Link(),
+						},
+					)
```

Comment:
to reviewer: need to validate this is the correct issuer, audience, and resource. Should one or more of these be the upload service?